src/utils/googleAnalytics.js

```diff
@@ -0,0 +1,61 @@
+// @flow
+import React, { useEffect, useState } from 'react'
+import GoogleAnalytics from 'react-ga'
+import { getGoogleAnalyticsTrackingID } from '~/config'
+import { COOKIES_KEY } from '~/logic/cookies/model/cookie'
+import type { CookiesProps } from '~/logic/cookies/model/cookie'
+import { loadFromCookie } from '~/utils/cookies'
+import type { RouterProps } from '~/routes/safe/store/selectors'
+
+const trackingID = getGoogleAnalyticsTrackingID()
+
+if (!trackingID) {
+  console.error('[GoogleAnalytics] - In order to use google analytics you need to add an trackingID')
+} else {
+  GoogleAnalytics.initialize(trackingID)
+}
+
+
+const withTracker = (WrappedComponent, options = {}) => {
+  const [useAnalytics, setUseAnalytics] = useState(false)
+
+  useEffect(() => {
+    async function fetchCookiesFromStorage() {
+      const cookiesState: CookiesProps = await loadFromCookie(COOKIES_KEY)
+      if (cookiesState) {
+        const { acceptedAnalytics } = cookiesState
+        setUseAnalytics(acceptedAnalytics)
+      }
+    }
+    fetchCookiesFromStorage()
+  }, [])
+
+  const trackPage = (page) => {
+    if (!useAnalytics) {
+      return
+    }
+    GoogleAnalytics.set({
+      page,
+      ...options,
+    })
+    GoogleAnalytics.pageview(page)
+  }
+
+  const HOC = (props: RouterProps) => {
+    // eslint-disable-next-line react/prop-types
+    const { location } = props
+    useEffect(() => {
+      const page = location.pathname + location.search
+      trackPage(page)
+    }, [])
+    useEffect(() => {
+      const page = location.pathname + location.search
+      trackPage(page)
+    }, location.pathname)
```

Comment:
I think having just 
```
    useEffect(() => {
      const page = location.pathname + location.search
      trackPage(page)
    }, [location.pathname])
```
is enough. This will fire on mount and every location.pathname change. Also useEffect's dependencies should be in an array

Reply:
You are right thanks!, got a typo there, now it's fixed

Reply:
💯thanks